spec/models/user_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  subject { build(:user, last_name: 'Doe', first_name: 'John', role: 'customer') }
+
+  context 'when valid Factory' do
```

Comment:
context can be omitted: you shouldn't validate the factory in the model spec or any other place. also, you aren't really doing that(you are validating the subject)

so, just
```
it { is_expected.to be_valid }
```